File: careerswipe-app/src/utils/openrouter.ts
export async function optimizeResume(resume: string, job: string): Promise<string> {
  const apiKey = process.env.NEXT_PUBLIC_OPENROUTER_API_KEY;
  if (!apiKey) {
    console.warn('OpenRouter API key missing');
    return resume;
  }
  try {
    const response = await fetch('https://openrouter.ai/api/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${apiKey}`,
      },
      body: JSON.stringify({
        model: 'gpt-3.5-turbo',
        messages: [
          { role: 'system', content: 'You are a helpful assistant that optimizes resumes for job descriptions.' },
          { role: 'user', content: `Job Description:\n${job}` },
          { role: 'user', content: `Resume:\n${resume}` },
          { role: 'user', content: 'Return the improved resume.' },
        ],
      }),
    });
    const data = await response.json();
    return data.choices?.[0]?.message?.content || resume;
  } catch (err) {
    console.error(err);
    return resume;
  }
}

File: backend/openrouter.py
import os
import requests

OPENROUTER_URL = "https://openrouter.ai/api/v1/chat/completions"


def ask_openrouter(messages):
    """Send chat messages to OpenRouter API and return response text."""
    api_key = os.environ.get("OPENROUTER_API_KEY")
    if not api_key:
        raise RuntimeError("OPENROUTER_API_KEY environment variable not set")
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    }
    payload = {
        "model": "openrouter/gpt-3.5-turbo",  # example model
        "messages": messages,
    }
    resp = requests.post(OPENROUTER_URL, headers=headers, json=payload, timeout=30)
    resp.raise_for_status()
    data = resp.json()
    return data["choices"][0]["message"]["content"]
